components/brave_wallet_ui/common/trezor/trezor_bridge_keyring.ts

```diff
@@ -0,0 +1,200 @@
+
+import { getLocale } from '../../../common/locale'
+
+const ethUtil = require('ethereumjs-util')
+
+export default class TrezorBridgeKeyring extends EventEmitter {
+  constructor () {
+    super()
+    this.unlocked_ = false
+  }
+
+  type = () => {
+    return kTrezorHardwareVendor
+  }
+
+  getAccounts = (from: number, to: number, scheme: string) => {
+    return new Promise(async (resolve, reject) => {
+      if (from < 0) {
+        from = 0
+      }
+      try {
+        if (!this.isUnlocked() && !(await this.unlock())) {
+          return reject(new Error(getLocale('braveWalletUnlockError')))
+        }
+      } catch (e) {
+        reject(e)
+        return
+      }
+      this._getAccounts(from, to, scheme).then(resolve).catch(reject)
+    })
+  }
+
+  isUnlocked = () => {
+    return this.unlocked_
+  }
+
+  unlock = () => {
+    return new Promise(async (resolve, reject) => {
+      let bridge = document.getElementById(kTrezorBridgeFrameId) as HTMLIFrameElement
+      if (!bridge) {
+        bridge = await this._createBridge() as HTMLIFrameElement
+      }
+      const kUnlockEvent = { command: kTrezorUnlockCommand, id: new Date().getTime(), owner: window.origin }
+      if (!bridge.contentWindow) {
+        throw Error(getLocale('braveWalletCreateBridgeError'))
+      }
+      this.addEventListener(kUnlockEvent.id, (data: any) => {
+        this.unlocked_ = data.result
+        if (data.result) {
+          resolve(true)
+        } else {
+          reject(false)
+        }
+      })
+      bridge.contentWindow.postMessage(kUnlockEvent, kTrezorBridgeUrl)
+    })
+  }
+
+  addWindowMessageListener = () => {
+    window.addEventListener('message', this.onMessageReceived.bind(this))
+  }
+
+  removeWindowMessageListener = () => {
+    window.removeEventListener('message', this.onMessageReceived.bind(this))
+  }
+
+  addEventListener = (id: number, listener: Function) => {
+    if (!this.pending_requests_) {
+      this.addWindowMessageListener()
+      this.pending_requests_ = {}
+    }
+    if (this.pending_requests_.hasOwnProperty(id)) {
+      return false
+    }
+    this.pending_requests_[id] = listener
+    return true
+  }
+
+  removeEventListener = (id: number) => {
+    if (!this.pending_requests_.hasOwnProperty(id)) {
+      return false
+    }
+    delete this.pending_requests_[id]
+    if (!Object.keys(this.pending_requests_).length) {
+      this.pending_requests_ = null
+      this.removeWindowMessageListener()
+    }
+    return true
+  }
+
+  onMessageReceived = (event: any) => {
+    if (event.origin !== kTrezorBridgeUrl || event.type !== 'message') {
+      return
+    }
+    if (!event.data || !this.pending_requests_ ||
+        !this.pending_requests_.hasOwnProperty(event.data.id)) {
+      return
+    }
+    const callback = this.pending_requests_[event.data.id] as Function
+    callback.call(this, event.data)
+    this.removeEventListener(event.data.id)
+  }
+
+  _getBridge = () => {
```

Comment:
Please use `private` keyword for methods instead of naming convention

Reply:
done